landing_page/lib/tasks/archive_posts.rake

```diff
@@ -0,0 +1,12 @@
+desc "Archive posts periodically based on if they are one week old or more"
+
+task archive: :environment do
+    archiveable_posts = Micropost.order(created_at: :desc).kept.where("created_at <= ?", Time.now-7.days)
+
+    archiveable_posts.each do |archiveable_post|
+        archiveable_post.archived = true
+        archiveable_post.save
```

Comment:
This is not a blocker, but rather a suggestion for you to get to know. Alternatively, instead of setting one or more attributes each line and then calling the `save` method, you can use the following method, that automatically saves all the changes:

```suggestion
        archiveable_post.update_attributes(archived: true)
```

You can have a look at the documentation [here](https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update_attributes)

Reply:
I'll have a look into it, thanks 👍